.github/workflows/build_cmake.yml

```diff
@@ -44,24 +53,52 @@ jobs:
 
     - name: Install conan
       run: |
-        pip3 install wheel setuptools
+        pip3 install wheel setuptools gcovr
         pip3 install conan --upgrade
-    - name: Ubuntu - Fix Conan Path
-      if: startsWith(matrix.os,'ubuntu')
+
+    - name: Install / Configure dependencies
+      shell: bash
       run: |
-        sudo update-alternatives --install /usr/bin/conan conan /home/runner/.local/bin/conan 10
-        sudo update-alternatives --config conan
+        if [ "$RUNNER_OS" == "Linux" ]; then
+          echo "Ubuntu - Fix Conan Path"
+          sudo update-alternatives --install /usr/bin/conan conan /home/runner/.local/bin/conan 10
+          sudo update-alternatives --config conan
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+          echo "Using chocolatey to install OpenCppCoverage"
+          choco install opencppcoverage
+          # Add to Path
+          echo "C:/Program Files/OpenCppCoverage" >> $GITHUB_PATH
+        # elif [ "$RUNNER_OS" == "macOs" ]; then
+        fi;
 
     - name: Configure CMake
       run: |
-        cmake -S . -B ./build -DCMAKE_BUILD_TYPE=$BUILD_TYPE
+          cmake -S . -B ./build -DCMAKE_BUILD_TYPE:STRING=$BUILD_TYPE -DENABLE_COVERAGE:BOOL=${{ matrix.ENABLE_COVERAGE }}
 
     - name: Build
       # Execute the build.  You can specify a specific target with "--target <NAME>"
       run: cmake --build ./build --config $BUILD_TYPE
 
-    - name: Test
+    - name: Unix - Test and coverage
+      if: runner.os != 'Windows'
       working-directory: ./build
       # Execute tests defined by the CMake configuration.
       # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-      run: ctest -C $BUILD_TYPE
+      run: |
+          set -x
```

Comment:
Is this required?
```suggestion
```

Reply:
I don't think that is useful unless you want to debug something. This also makes the code dependant on Bash.

Reply:
Bash is available on all three platforms: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

If you add the line “shell: bash”, it makes the dependency explicit, and overrides the Windows default shell.

Reply:
Bash may not be installed in self-hosted runners or distro-less docker containers.

Reply:
You make a pretty good point. It's sloppy to add `set -x` to this block but not every other block of bash code. Furthermore, if we add `set -x` to every block of bash, the workflow becomes harder to adapt for self-hosted runners or docker containers.

I think that adding `set -x` to all the bash code is potentially a good idea, but it solves a different problem than this PR is meant to, so I think it's out of scope and should be evaluated separately.

Reply:
I don't understand why it's sloppy. If you don't care about individual commands inside a step, then don't use -x, if you do then why not?
anyways, I don't care.

Reply:
I’m sorry, I didn’t mean to imply that your code was sloppy, that was a poor choice of words. I meant that consistency should be important for this project. It’s a good idea to add “set -x”, and we should seriously consider adding “set -x” to all the bash code in CI. However, I don’t think we need to let disagreements about “set -x” derail this PR.

Thanks for your hard work, @jmarrec